Source/FieldSolver/ElectrostaticSolver.cpp

```diff
@@ -137,41 +137,36 @@ WarpX::AddSpaceChargeFieldLabFrame ()
                                      "Error: RZ electrostatic only implemented for a single mode");
 #endif
 
-    // Allocate fields for charge
-    const int num_levels = max_level + 1;
-    Vector<std::unique_ptr<MultiFab> > rho(num_levels);
-    // Use number of guard cells used for local deposition of rho
-    const amrex::IntVect ng = guard_cells.ng_depos_rho;
+    // Allocate fields for the potential
+    // Also, zero out the phi data
     for (int lev = 0; lev <= max_level; lev++) {
         BoxArray nba = boxArray(lev);
         nba.surroundingNodes();
-        rho[lev] = std::make_unique<MultiFab>(nba, dmap[lev], 1, ng);
-        rho[lev]->setVal(0.);
+        rho_fp[lev]->setVal(0.);
     }
 
     // Deposit particle charge density (source of Poisson solver)
     for (int ispecies=0; ispecies<mypc->nSpecies(); ispecies++){
         WarpXParticleContainer& species = mypc->GetParticleContainer(ispecies);
         bool const local = true;
+        bool const interpolate_across_levels = false;
         bool const reset = false;
         bool const do_rz_volume_scaling = false;
-        species.DepositCharge(rho, local, reset, do_rz_volume_scaling);
-    }
-    for (int lev = 0; lev <= max_level; lev++) {
-        ApplyFilterandSumBoundaryRho (lev, lev, *rho[lev], 0, 1);
+        species.DepositCharge(rho_fp, local, reset, do_rz_volume_scaling, interpolate_across_levels);
     }
 #ifdef WARPX_DIM_RZ
     for (int lev = 0; lev <= max_level; lev++) {
-        ApplyInverseVolumeScalingToChargeDensity(rho[lev].get(), lev);
+        ApplyInverseVolumeScalingToChargeDensity(rho_fp[lev].get(), lev);
```

Comment:
A small clean up would be to move this statement into the previous loop over levels. This doesn't need its own loop.

Reply:
Hm, the thing is that the loop over species (for the associated charge deposition) is needed in between these 2 loops over levels. So I am not sure how to merge the loops over levels. Did I miss something?

Reply:
Yes, you are correct. I was confused by the way the changes are shown.